qiskit_experiments/library/characterization/fine_frequency.py

```diff
@@ -0,0 +1,129 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2021.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+"""Fine frequency characterization experiment."""
+
+from typing import List, Optional
+import numpy as np
+
+from qiskit import QuantumCircuit
+from qiskit.providers.backend import Backend
+
+from qiskit_experiments.framework import BaseExperiment, Options
+from qiskit_experiments.library.characterization.analysis import FineFrequencyAnalysis
+
+
+class FineFrequency(BaseExperiment):
+    r"""An experiment to make a fine measurement of the qubit frequency.
+
+    # section: overview
+
+        The fine frequency characterization experiment measures the qubit frequency by moving
+        to the equator of the Bloch sphere with a sx gate and idling for a time
+        :math:`n\cdot \tau` where :math:`\tau` is the duration of the single-qubit gate and
+        :math:`n` is an integer which ranges from zero to a maximal value in integer steps.
+        The idle time is followed by a rz rotation with an angle :math:`n\pi/2` and a final
+        sx gate. If the frequency of the pulses match the frequency of the qubit then the
+        sequence :math:`n\in[0,1,2,3,4,...]` will result in the sequence of measured qubit
+        populations :math:`[1, 0.5, 0, 0.5, 1, ...]` due to the rz rotation. If the frequency
+        of the pulses do not match the qubit frequency then the qubit will precess in the
+        drive frame during the idle time and phase errors will build-up. By fitting the measured
+        points we can extract the error in the qubit frequency. The circuit that are run are
+
+        .. parsed-literal::
+
+                    ┌────┐┌────────────────┐┌──────────┐┌────┐ ░ ┌─┐
+                 q: ┤ √X ├┤ Delay(n*τ[dt]) ├┤ Rz(nπ/2) ├┤ √X ├─░─┤M├
+                    └────┘└────────────────┘└──────────┘└────┘ ░ └╥┘
+            meas: 1/══════════════════════════════════════════════╩═
+                                                                  0
+    # section: analysis_ref
+        :py:class:`FineFrequencyAnalysis`
+    """
+
+    def __init__(
+        self,
+        qubit: int,
+        delay_duration: int,
+        backend: Optional[Backend] = None,
+        repetitions: Optional[List[int]] = None,
+    ):
+        """Setup a fine frequency experiment on the given qubit.
+
+        Args:
+            qubit: The qubit on which to run the fine frequency characterization experiment.
+            delay_duration: The duration of the delay at :math:`n=1`.
```

Comment:
```suggestion
            delay_duration: The duration of the delay at :math:`n=1` in dt.
```
Probably we can set default value to 160? Is this too IBM-ish?

Reply:
I think this is too IBM-ish it's even AWG specific. My solution is to require the delay length as input in the characterization experiment and the calibration version will default to the duration of the sx schedule in the cals (unless the user provides the delay length or another schedule name).

Reply:
Another option would be to have an extra option for duration in seconds and convert that dt if no duration in dt is given. The default would still have to be biased to a particular architecture (similar to the defaults for T1 and T2).